src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -316,14 +317,18 @@ private async Task LoadKeyValuesRegisteredForRefresh(IDictionary<string, Configu
             {
                 string watchedKey = changeWatcher.Key;
                 string watchedLabel = changeWatcher.Label;
+                KeyLabelIdentifier watchedKeyLabel = new KeyLabelIdentifier(watchedKey, watchedLabel);
 
                 // Skip the loading for the key-value in case it has already been loaded
-                if (data.ContainsKey(watchedKey) && data[watchedKey].Label == watchedLabel.NormalizeNull())
+                if (data.ContainsKey(watchedKey) 
+                    && string.Equals(data[watchedKey].Key, watchedKey, StringComparison.Ordinal)
+                    && string.Equals(data[watchedKey].Label, watchedLabel.NormalizeNull(), StringComparison.Ordinal))

```

Comment:
We could use `==` operator to maintain consistency with existing code. It performs ordinal comparison by default.